src/utils/dif.rs

```diff
@@ -131,17 +130,34 @@ impl fmt::Display for DifFeatures {
     }
 }
 
+pub struct SelfProguard<'a>(ProguardMapping<'a>);
+
+impl<'a> std::ops::Deref for SelfProguard<'a> {
+    type Target = ProguardMapping<'a>;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<'slf> symbolic::common::AsSelf<'slf> for SelfProguard<'_> {
+    type Ref = SelfProguard<'slf>;
+
+    fn as_self(&'slf self) -> &Self::Ref {
+        self
+    }
+}
+
 pub enum DifFile<'a> {
     Archive(SelfCell<ByteView<'a>, Archive<'a>>),
-    Proguard(ProguardMappingView<'a>),
+    Proguard(SelfCell<ByteView<'a>, SelfProguard<'a>>),
 }
 
 impl DifFile<'static> {
     fn open_proguard<P: AsRef<Path>>(path: P) -> Result<Self, Error> {
-        let data = ByteView::open(&path).map_err(SyncFailure::new)?;
-        let pg = ProguardMappingView::parse(data).map_err(SyncFailure::new)?;
+        let data = ByteView::open(path).map_err(SyncFailure::new)?;
+        let pg = SelfCell::new(data, |d| SelfProguard(ProguardMapping::new(unsafe { &*d })));
 
-        if path.as_ref().extension() == Some(OsStr::new("txt")) || pg.has_line_info() {
```

Comment:
this is technically a change in behavior. Before this change, the cli would erroneously flag *any* `.txt` file as proguard. Not you will get a `Unsupported file` error.

Reply:
I think that's fine though